storage/database/shardedPersister_test.go

```diff
@@ -0,0 +1,387 @@
+		benchmarkPersister(b, keys, shardedPersisterPath, "sharded")
+	})
+}
+
+func BenchmarkPersister1mil10kKeys(b *testing.B) {
+	entries, keys := generateKeys(1000000)
+
+	persisterPath := b.TempDir()
+	_ = createAndPopulatePersister(persisterPath, entries)
+	shardedPersisterPath := b.TempDir()
+	_ = createAndPopulateShardedPersister(shardedPersisterPath, entries)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, keys[0:10000], persisterPath, "simple")
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, keys[0:10000], shardedPersisterPath, "sharded")
+	})
+}
+
+func BenchmarkPersister1mil100kKeys(b *testing.B) {
+	entries, keys := generateKeys(1000000)
+
+	persisterPath := b.TempDir()
+	_ = createAndPopulatePersister(persisterPath, entries)
+	shardedPersisterPath := b.TempDir()
+	_ = createAndPopulateShardedPersister(shardedPersisterPath, entries)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, keys[0:100000], persisterPath, "simple")
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, keys[0:100000], shardedPersisterPath, "sharded")
+	})
+}
+
+func BenchmarkPersister2milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(2000000)
+
+	persisterPath := b.TempDir()
+	_ = createAndPopulatePersister(persisterPath, entries)
+	shardedPersisterPath := b.TempDir()
+	_ = createAndPopulateShardedPersister(shardedPersisterPath, entries)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, persisterPath, "simple")
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, shardedPersisterPath, "sharded")
+	})
+}
+
+func BenchmarkPersister4milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(4000000)
+
+	persisterPath := b.TempDir()
+	_ = createAndPopulatePersister(persisterPath, entries)
+	shardedPersisterPath := b.TempDir()
+	_ = createAndPopulateShardedPersister(shardedPersisterPath, entries)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, persisterPath, "simple")
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, shardedPersisterPath, "sharded")
+	})
+}
+
+func BenchmarkPersister8milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(8000000)
+
+	persisterPath := b.TempDir()
+	_ = createAndPopulatePersister(persisterPath, entries)
+	shardedPersisterPath := b.TempDir()
+	_ = createAndPopulateShardedPersister(shardedPersisterPath, entries)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, persisterPath, "simple")
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, keys, shardedPersisterPath, "sharded")
+	})
+}
+
+func createPersister(path string, id string) (storage.Persister, error) {
+	switch id {
+	case "simple":
+		return NewLevelDB(path, 2, 1000, 1000)
+	case "sharded":
+		shardCoordinator, _ := sharding.NewMultiShardCoordinator(3, 0)
```

Comment:
please use a power of 2 for the number of shards, e.g 4. 
otherwise some shards will be more used than others.
In this case with 3 shards, in shard 1 you (statistically) have 2x the number of keys/values from shard  0 or shard 1